### PR TITLE
THORN-2064, update to use smallrye-jwt 1.1.0 implementation

### PR DESCRIPTION
**Pull request by starksm64**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [x ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: fractions/microprofile/microprofile-jwt/pom.xml [hunk 1]
```diff
@@ -36,6 +37,18 @@
     </resources>
   </build>
 
+  <dependencyManagement>
```
Comment (sberyozkin) [inline fractions/microprofile/microprofile-jwt/pom.xml]: Why is Arquillian dependency needed in the fraction pom ?

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java [hunk 2]
```diff
@@ -208,6 +210,7 @@ void processClaimProviderInjections(@Observes ProcessInjectionPoint<?, ? extends
      */
     void observesAfterBeanDiscovery(@Observes final AfterBeanDiscovery event, final BeanManager beanManager) {
         log.debugf("observesAfterBeanDiscovery, %s", claims);
+        //event.addBean(new JWTPrincipalBean());
```
Comment (sberyozkin) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java]: The commented line can probably go

**Comment by sberyozkin** [comment]
I've done a couple of comments, it looks good from my non-expert point of view. I'd like to ask Ladislav to give it a good look as well :-), and then I'll merge

File: fractions/microprofile/microprofile-jwt/pom.xml (hunk 1 above)
Comment (starksm64) [reply fractions/microprofile/microprofile-jwt/pom.xml]: It was used during some initial testing, but has been removed now.

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java (hunk 2 above)
Comment (starksm64) [reply fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java]: Updated

**Comment by kenfinnigan** [comment]
Is this ready for integration?

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java [hunk 3]
```diff
@@ -116,6 +117,7 @@ public void afterDeploymentValidation(@Observes AfterDeploymentValidation event,
     }
 
     void doProcessProducers(@Observes ProcessProducer pp) {
+        log.infof("doProcessProducer, pp=%s", pp.getProducer());
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/MPJWTExtension.java]: `debug` at worst, preferrably `trace`.

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/PrincipalProducer.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import javax.annotation.Priority;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Produces;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+/**
+ * Override the
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/cdi/PrincipalProducer.java]: ??

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java [hunk 5]
```diff
@@ -20,28 +20,34 @@
...
+import io.smallrye.jwt.KeyUtils;
+import io.smallrye.jwt.auth.principal.JWTAuthContextInfo;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
-import org.wildfly.swarm.microprofile.jwtauth.deployment.auth.KeyUtils;
-import org.wildfly.swarm.microprofile.jwtauth.deployment.principal.JWTAuthContextInfo;
+import org.jboss.logging.Logger;
 
 /**
- *
+ * An extension of the Smallrey CDI provider for the JWTAuthContextInfo that extends the information from
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java]: SmallRye

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java [hunk 6]
```diff
@@ -20,28 +20,34 @@
...
+ * MP-JWT config properties with legacy fraction properties as well as properties for features not
+ * convered by MP-JWT settings.
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java]: "covered"?

File: fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java [hunk 7]
```diff
@@ -52,36 +58,82 @@
...
+            // If there is no expected issuer configured, don't validate it; new in MP-JWT 1.1
+            contextInfo.setRequireIssuer(false);
         }
+
+        Optional<Boolean> mpJwtRequireIss = super.getMpJwtRequireIss();
+        if (mpJwtRequireIss != null && mpJwtRequireIss.isPresent()) {
+            contextInfo.setRequireIssuer(mpJwtRequireIss.get());
+        } else {
+            // Default is to require iss claim
+            contextInfo.setRequireIssuer(true);
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java:122]: A couple lines above, we have

```java
// If there is no expected issuer configured, don't validate it; new in MP-JWT 1.1
contextInfo.setRequireIssuer(false);
```

And here, we do `contextInfo.setRequireIssuer(true);`, both as a fallback if the respective property is not configured. That looks strange to me.

**Comment by Ladicek** [comment]
Added a couple of small comments. Also a full CI build would be nice.

**Comment by kenfinnigan** [comment]
retest this please

**Comment by kenfinnigan** [comment]
How is this progressing @starksm64?

**Comment by starksm64** [comment]
I'm trying to track down what is causing the tests to not even start up. Some dependency looks to be missing as the swarm.application module is not being loaded.

```
Sun Jul 15 23:07:26 PDT 2018 INFO [org.wildfly.swarm.bootstrap] (main) Dependencies not bundled; resolving from M2REPO.
java.lang.ClassNotFoundException: org.wildfly.swarm.arquillian.adapter.AnnotationBasedMain
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.wildfly.swarm.bootstrap.MainInvoker.getMainClass(MainInvoker.java:117)
	at org.wildfly.swarm.bootstrap.MainInvoker.<init>(MainInvoker.java:49)
	at org.wildfly.swarm.bootstrap.Main.run(Main.java:131)
	at org.wildfly.swarm.bootstrap.Main.main(Main.java:85)
Exception in thread "main" java.lang.ClassNotFoundException: org.wildfly.swarm.arquillian.adapter.AnnotationBasedMain
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.wildfly.swarm.bootstrap.MainInvoker.getMainClass(MainInvoker.java:117)
	at org.wildfly.swarm.bootstrap.MainInvoker.<init>(MainInvoker.java:49)
	at org.wildfly.swarm.bootstrap.Main.run(Main.java:131)
	at org.wildfly.swarm.bootstrap.Main.main(Main.java:85)
```

**Comment by kenfinnigan** [comment]
Maybe try rebasing onto the latest, as there may have been a change somewhere that is causing an issue?

`io.thorntail:arquillian` is definitely in the pom of the testsuite, so only thing I can think is for some reason it's not looking for that one?

**Comment by starksm64** [comment]
I had merged the upstream changes previously, and remerged them yesterday.

The issue was some problem with the org.wildfly.swarm.configuration.microprofile.config module not be resolved despite there being a dependency on the io.thorntail:config fraction. I ended up copying the metrics fraction dependencies as that seemed to worked.

**Comment by starksm64** [comment]
As of the d07de5b commit, all of the tests are passing, so we'll see how the CI build looks:

JWT:
```
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:24 min
[INFO] Finished at: 2018-07-17T06:27:55-07:00
[INFO] ------------------------------------------------------------------------
```

TCK:
```
[INFO] Tests run: 113, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 261.579 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 113, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:27 min
[INFO] Finished at: 2018-07-17T06:24:25-07:00
[INFO] ------------------------------------------------------------------------
```

**Comment by kenfinnigan** [comment]
Looks like CI is all good.

I started taking a look at the changes to review and it looks like the rebase went funny, as there are changes in /docs and other places that I wouldn't expect as part of this PR. Could you take a look?

**Comment by starksm64** [comment]
I'll redo the PR at this point because things seem strange between my local repo, origin and upstream.

**Pull request closed without merging**